### PR TITLE
(Part 1) Integrate Assignment Service With Frontend (Get Endpoints only)

### PR DESCRIPTION
**Pull request by tryyang2001**
# Pull Request

## Description

I am planning to do refactoring for frontend code, so it will involve a massive changes, hence I decided to separate the PRs. For now, I have integrated the frontend /dashboard page, and /assignments/:id page with the assignment service GET endpoints. I will continue working on the create assignment page, and also create a page for creating questions, test cases, reference solution, as well as updating them. 

In this PR, I also reroute /assignments/dashboard page to /dashboard. This should be the home page when one logs in.

Last but not least, I also added a node script called "dev-all", which will run all the microservices in one command. 

## Related Issue(s)

This should be part of the work for integrating assignment service with the frontend.

## Screenshots (if applicable)

image
The screenshot for /dashboard page. This is the actual available assignments in the database.

image
When clicking the `New Attempt` button, it will navigate you to the individual assignment page, with id shows in the path. 

## Checklist

- [x] I have checked that the changes included in the PR are intended to merge to `master` or any destination branch.
- [x] I have verified that the new changes do not break any existing functionalities, unless the new changes are intended and have approved by the team.
- [x] I will take care of the merging and delete the side-branch after the PR is merged.

## Additional Notes/References

Many things to take note:
1. @hhchinh2002 We need user schema to have an "assignments" field that will tell for each user, what assignments he or she is involved. In the future, when we create the course service, this "assignments" field should be replaced by "course", so I am not sure if you want to do it now, or wait until we have the course service. This is very important as without this field, I don't know what assignments to show :(
2. @junhonglow I will refactor the assignment creation page, and add more input fields FYI. Also, I will create more forms for updating/creating assignment/question details.
3. @vivienherq I noticed that there are 2 scroll bars (see example above) due to the "Log Out" button at the bottom of the side bar, perhaps you might want to take a look.

### CODE REVIEW

File: frontend/src/app/assignments/[id]/layout.tsx [hunk 1]
```diff
@@ -16,11 +16,11 @@ export default function AssignmentPageLayout({
...
-  const assignmentTitle = (
-    await AssignmentService.getAssignmentById({
-      assignmentId: params.id,
-    })
-  ).title;
+  const currentAssignment = await AssignmentService.getAssignmentById(
+    params.id
+  );
+
+  const assignmentTitle = currentAssignment?.title ?? "No such assignment";
```
Comment (junhonglow) [inline frontend/src/app/assignments/[id]/layout.tsx:23]: Are we expecting the page to 404 atm if an assignment is not found?
I see that you made some logic to handle if the assignment is not found here.

**Review by junhonglow — COMMENTED** [review]
Logic looks great just some places I need to clarify on.

> In this PR, I also reroute /assignments/dashboard page to /dashboard. This should be the home page when one logs in.

I think this change hasn't been pushed yet.

File: frontend/src/app/assignments/[id]/layout.tsx (hunk 1 above)
Comment (tryyang2001) [reply frontend/src/app/assignments/[id]/layout.tsx:23]: Yes, I don't think we need to focus on the error page for now.
For this, I am just controlling the page title shown in the browser tab.

**Review by junhonglow — APPROVED** [review]

**Comment by vivienherq** [comment]
can i check if students will submit one file (containing submissions for all questions) for each assignment? or would they be submitting one file for each question? if it is the former, how will the code for each question be differentiated and stored?

**Comment by tryyang2001** [comment]
They will submit files per question. Currently, they can submit unlimited times for each question, so we need to limit the number of submission in the future...
The submitted code will be stored in "Submission" table in the grading service database, to differentiate each submission, we can use "userId" and "questionId", I have created an endpoint in grading service called [GET /questions/:id/submissions](https://github.com/tryyang2001/CS3213-Frontend-Management-System/blob/master/backend/grading-service/src/routes/route.ts#L13) for retrieving the submission [given a student id](https://github.com/tryyang2001/CS3213-Frontend-Management-System/blob/master/backend/grading-service/src/controllers/grading-controller.ts#L23) in the query parameter.

**Pull request merged**